plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -6708,47 +6715,162 @@ public void testAlterTableWithUnsupportedProperties()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testDropTableWithMissingMetadataFile()
+            throws Exception
+    {
+        String tableName = "test_drop_table_with_missing_metadata_file_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 x, 'INDIA' y", 1);
+
+        TrinoFileSystem trinoFileSystem = fileSystemFactory.create(SESSION);
+        String tableLocation = getTableLocation(tableName);
+        String metadataLocation = getLatestMetadataLocation(trinoFileSystem, tableLocation);
+
+        // Delete current metadata file
+        trinoFileSystem.deleteFile(metadataLocation);
+        assertFalse(trinoFileSystem.newInputFile(metadataLocation).exists(), "Current metadata file should not exist");
+
+        // try to drop table
+        assertUpdate("DROP TABLE " + tableName);
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
+        assertFalse(trinoFileSystem.listFiles(tableLocation).hasNext(), "Table location should not exist");
+    }
+
+    @Test
+    public void testDropTableWithMissingSnapshotFile()
```

Comment:
We should also make sure that we don't delete on `DROP` more than the content associated to the corrupted table.

I'm thinking that a test where two tables exist before the drop:

- one table is ok 
- one table is corrupt

When dropping the corrupt table, the OK table should still be present in the metastore and the amount of files in the storage (within the test schema) should decrease with only the number of files corresponding to the corrupted table.

Reply:
having the test as part of BCT / BCST provides that. there are other test tables (like nation) which don't get deleted.